src/main/java/org/terasology/launcher/LauncherInitTask.java

```diff
@@ -259,6 +272,27 @@ private Path getGameDataDirectory(OperatingSystem os, Path settingsGameDataDirec
         return gameDataDirectory;
     }
 
+    /**
+     * Shows a confirmation dialog for overwriting current sources file
+     * with default values.
+     *
+     * @return whether the user confirms this overwrite
+     */
+    private boolean confirmSourcesOverwrite() {
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
👍 this is pretty nice - maybe we should adapt our code in `GuiUtils` to follow a similar pattern and get rid of custom handling of tasks and the like.